linux/ibus-keyman/build.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+set -eu
+
+## START STANDARD BUILD SCRIPT INCLUDE
+# adjust relative paths as necessary
+THIS_SCRIPT="$(greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null || readlink -f "${BASH_SOURCE[0]}")"
+. "$(dirname "$THIS_SCRIPT")/../../resources/build/build-utils.sh"
+## END STANDARD BUILD SCRIPT INCLUDE
+
+################################ Main script ################################
+
+builder_describe \
+  "Build ibus-keyman." \
+  "clean" \
+  "configure" \
+  "build" \
+  "test" \
+  ":engine" \
+  "default+                  clean, configure, and build" \
+  "install                   install artifacts" \
+  "uninstall                 uninstall artifacts" \
+  "--debug,-d                Debug build" \
+  "--output=OUTPUT_PATH,-o   Output path (default: ../build/)"
+
+builder_parse "$@"
+
+if builder_has_action default; then
+  _builder_chosen_action_targets=("clean:engine" "configure:engine" "build:engine")
```

Comment:
If `:engine` is the only target, do we still need to specify it?

Reply:
I tinkered with something like this at one point - where enabling one action or target (I forget which) would flag and activate other actions or targets as if they'd be specified.  If I recall correctly, we decided to go a different way in those cases.  I'd almost gotten to the point of adding a builder function (called `builder_set_action` or similar) that script could use for such activation.

Given the fact we've actively avoided this so far... feels kind of odd to maybe go back on it now.  But that's just my take.

Also... it's not like `clean configure build` is a hard set of actions to remember - it's _very_ consistent among our scripts thus far.

At the same time, I'm not opposed to a special keyword `all` - an action that simply says "activate all action:target groupings".

Reply:
It might be better to solve this with dependencies in which case we don't necessarily need a `linux/build.sh` script. On the other hand it's convenient to be able to run a command in one location to build everything that's needed for a platform.

IMO that makes it easier for on-boarding, and reduces the risk of stale documentation in regards to the build process ("Oops, we forgot to add that you now have to build that other project as well...").

Reply:
Yes, dependencies are designed to allow a first-build with just `./build.sh configure build [test] [install]`... A rebuild of a dependency should only be needed if that dependency changes (e.g. `git pull` or branch change). Eventually, we'll have a top level script which will allow a `./build.sh clean configure:linux build:linux` which would cleanup all projects and then rebuild linux (and we may be able to tweak that command also to `./build.sh rebuild:linux` or similar).

Two intents:

1. Reduce [re]build time.
2. Simplify instructions.